test/puppetlabs/puppetdb/threadpool_test.clj

```diff
@@ -32,14 +32,39 @@
 
       (with-open [gtp (gated-threadpool 1 "test-pool-%d" 1000)]
         (let [fut (future
-                    (dochan gtp worker-fn in-chan))]
+                    (dochan gtp worker-fn in-chan tu/blocked?))]
           (async/>!! in-chan {:id :a})
           (is (= [:done :a] (async/<!! out-chan)))
           (is (= 1 @counter))
           (async/close! in-chan)
           (is (not= ::timed-out (deref fut tu/default-timeout-ms ::timed-out)))
           (is (true? (future-done? fut)))))))
 
+
+  (testing "block command"
+    (let [counter (atom 0)
+          in-chan (async/chan 10)
+          out-chan (async/chan 10)
+          worker-fn (wrap-out-chan
+                      out-chan
+                      (fn [{:keys [id]}]
+                        (swap! counter inc)
+                        [:done id]))
+          blocked? (atom true)]
+
+        (with-open [gtp (gated-threadpool 1 "test-pool-%d" 1000)]
+          (let [fut (future
+                      (dochan gtp worker-fn in-chan blocked?))]
+            (async/>!! in-chan {:id :a})
+            (let [channel (future (async/<!! out-chan))
+                  ]
+              (is (= (deref channel 1000 "Did not wake up!") "Did not wake up!") )
+              )
+            (is (= 0 @counter))
+            (async/close! in-chan)
+            (is (= ::timed-out (deref fut 1000 ::timed-out)))
+            (is (true? (future-cancel fut)))))))
```

Comment:
I might lean toward just `(is (future-cancel fut))`.

Reply:
`(future-cancel fut)` was removed because now we unblock and the future can finish. Waiting for the future to finish was done with `(is (future-done? fut))`